sasmodels/sesanscosine.py

```diff
@@ -0,0 +1,214 @@
+"""
+Conversion of scattering cross section from SANS (I(q), or rather, ds/dO) in absolute
+units (cm-1)into SESANS correlation function G using a Hankel transformation, then converting
+the SESANS correlation function into polarisation from the SESANS experiment
+
+Everything is in units of metres except specified otherwise (NOT TRUE!!!)
+Everything is in conventional units (nm for spin echo length)
+
+Wim Bouwman (w.g.bouwman@tudelft.nl), June 2013
+"""
+
+from __future__ import division
+
+import numpy as np  # type: ignore
+from numpy import pi, exp  # type: ignore
+from scipy.special import jv as besselj
+#import direct_model.DataMixin as model
+        
+def make_q(q_max, Rmax):
+    r"""
+    Return a $q$ vector suitable for SESANS covering from $2\pi/ (10 R_{\max})$
+    to $q_max$. This is the integration range of the Hankel transform; bigger range and 
+    more points makes a better numerical integration.
+    Smaller q_min will increase reliable spin echo length range. 
+    Rmax is the "radius" of the largest expected object and can be set elsewhere.
+    q_max is determined by the acceptance angle of the SESANS instrument.
+    """
+    from sas.sascalc.data_util.nxsunit import Converter
+
+    q_min = dq = 0.1 * 2*pi / Rmax
+    return np.arange(q_min,
+                     Converter(q_max[1])(q_max[0],
+                                         units="1/A"),
+                     dq)
+    
+def make_all_q(data):
+    """
+    Return a $q$ vector suitable for calculating the total scattering cross section for
+    calculating the effect of finite acceptance angles on Time of Flight SESANS instruments.
+    If no acceptance is given, or unwanted (set "unwanted" flag in paramfile), no all_q vector is needed.
+    If the instrument has a rectangular acceptance, 2 all_q vectors are needed.
+    If the instrument has a circular acceptance, 1 all_q vector is needed
+    
+    """
+    if not data.has_no_finite_acceptance:
```

Comment:
is `.has_no_finite_acceptance` an attribute of all data classes - and should it be?